### DISCUSSION TITLE
Why can't `_basename` be used in function? Why must the `title` field exist to activate the processor?

### DISCUSSION THREAD

**Discussion by ooker777** [snezhig/obsidian-front-matter-title#226] [Q&A] [answered]
I want to have the title based on the filename, not the `title` field. To do that I must assign `_basename` as the fallback template, and use `title` in the function:
```js
if (title.slice(-1) === "？") return title.replace(/？$/g, "?")
```
Why can't I use this directly?
```js
if (_basename.slice(-1) === "？") return _basename.replace(/？$/g, "?")
```
The problem of having to use `basename` in as the fallback template is that if I set [note link](https://github.com/snezhig/obsidian-front-matter-title/blob/master/docs//Features/NoteLink.md "obsidian-front-matter-title/docs/Features/NoteLink.md at master · snezhig/obsidian-front-matter-title") to activate, then it will always return `[[basename|basename]]`. To have it not adding alias when there is no main template or fallback template, the fallback template must not be `_basename`.

**Comment by snezhig** [comment, marked as answer]
Hello.
`_basename` is an internal word for the plugin itself.
In the last update, a new processor was added that passes an object. There I can add another field with a TFile object from which you can get the basename.
